pkg/auth/providers/azure/clients/ms_graph_client_test.go

```diff
@@ -22,11 +22,38 @@ import (
 // TEST_AZURE_TENANT_ID
 // TEST_AZURE_APPLICATION_ID
 // TEST_AZURE_APPLICATION_SECRET
+// TEST_AZURE_CHINA
 
 func TestMSGraphClient_GetUser(t *testing.T) {
 	secrets := newTestSecretsClient()
 	client := newTestClientWithSecretsClient(t, secrets)
 
+	isTestAzureChina := os.Getenv("TEST_AZURE_CHINA")
```

Comment:
You could move the `isTestAzureChina` to a package level variable rather than each test, and make it a boolean? It won't change between tests?

```go
var isTestAzureChina = os.Getenv("TEST_AZURE_CHINA") == ""
```

Also, I'd be inclined to split the tests for China out into separate cases.

```go
func TestMSGraphClient_GetUser_China(t *testing.T) {
    if !isTestAzureChina {
      t.Skip("skipping China GetUser test")
    }
    secrets := newTestSecretsClient()
    client := newTestClientWithSecretsClient(t, secrets)

    user, err := client.GetUser("Jacie@rancherchina.partner.onmschina.cn")
    if err != nil {
        t.Fatal(err)
    }
    want := mgmtv3.Principal{
        PrincipalType: "user",
        Provider:      Name,
        ObjectMeta: metav1.ObjectMeta{
            Name: "azuread_user://7839f7fd-08a9-421e-b5a3-6931f6761ac9",
        },
        DisplayName: "Jacie",
        LoginName:   "Jacie@rancherchina.partner.onmschina.cn",
    }
    assert.Equal(t, want, user)

    client = newTestClientWithSecretsClient(t, secrets)
    _, err = client.GetUser("Jacie@rancherchina.partner.onmschina.cn")
    if err != nil {
        t.Fatal(err)
    }
}
```

Reply:
Thanks for the detail.